replica/table_check.cc

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "replica/database.hh"
+#include "replica/table_check.hh"
+#include "service/migration_manager.hh"
```

Comment:
Sorry, I just noticed this, I think this file should be in `repair/`. The first time I reviewed this I actually misread `replica` as `repair`.
This currently is a layering violation because the replica should not depend on the migration manager. We have many such layering violations but we shouldn't add new ones if not necessary.

Please move this to `repair/`.